src/Controllers/PoolProviderController.cs

```diff
@@ -338,7 +338,15 @@ public async Task<IActionResult> GetAgentRequestStatus([FromBody] AgentRequestSt
             // Need to know the job correlation id and work item id.
             // Work item id is the agent id, and job correlation is in the agent data.
             string workItemId = agentRequestStatusItem.agentId;
-            string correlationId = agentRequestStatusItem.agentData.correlationId;
+            string correlationId = agentRequestStatusItem.agentData?.correlationId;
+
+            if (string.IsNullOrEmpty(correlationId))
+            {
+                // This means the call came back before we've provisioned it.  We don't have any meaningful
+                // response right now, so return the empty string to reflect "no status".
+                _logger.LogInformation("Status request has no agentData, returning empty status");
+                return Json("");
```

Comment:
Should we return NoContent here?

Reply:
They said to return the empty string.  It's possible no-content will freak their system out.  Though honestly the only difference between the two is the number at the top.  Let me check with them and see.

Reply:
What's the http return code if we just return Json("")?

Reply:
Json is a wrapper around "Ok", which makes it 200.  "NoContent" is 204.  It shouldn't matter, both are "success". But sometimes people care a bit too much about the particular flavor of success. :-)

Reply:
Went with no-content. :-)